ci-operator/README.md

```diff
@@ -0,0 +1,154 @@
+images, executing unit tests, etc.  These can be built upon for e2e test,
+requiring little or no modification.
+
+After that, to add an e2e test:
+
+1. Determine the pre-requisites for the test.  Practically, this means choosing
+   the template that is going to be used by `ci-operator`.  This directory
+   already contains templates for the common cases, see
+   [Using a template](#using-a-template) below.
+2. Determine and configure the template's inputs.  This is specific to each
+   template and should be documented in its parameters.  The e2e test might
+   need minor modifications to fit the environment created by the template.
+3. Add one or more prow jobs to prow's configuration file with the information
+   gathered in the previous steps.
+
+
+# Provisioning a cluster
+
+Contrary to other types of tests, e2e tests usually require a cluster, not just
+a single container.  While there aren't yet native primitives in `ci-operator`
+for cluster provisioning, it provides one open-ended feature that can be
+leveraged to accomplish that: template steps.
+
+Template steps allow the creation of arbitrary objects in the cluster where the
+CI pipeline is executed.  This is used to start a pod that will then provision
+a separate cluster for the tests.  This directory already contains a few
+templates that can be used either directly or (rarely necessary, in practice)
+as a reference:
+
+- `cluster-launch-e2e.yaml`: launches a cluster in GCP using openshift-ansible
+  and runs Origin e2e tests on it, parameterized by test focus.
+- `cluster-launch-installer-e2e.yaml`:
+- `cluster-launch-src.yaml`: launches a cluster in GCP using openshift-ansible
+  and runs a script from the repository being tested with the resulting
+  `$KUBECONFIG`, parameterized by test script.
+- `master-sidecar.yaml`: spins up a simple openshift control plane as a sidecar
+  and waits for the `COMMAND` specified to the template to be executed, before
+  itself exiting. The test container is given access to the generated
+  configuration and the `admin.kubeconfig`.
+- `os.yaml`:
+
+
+# Using a template
+
+A template can be used as a stage in `ci-operator` with the `--template`
+argument.  The new template stage is added at runtime to those read from the
+configuration file.  The name of the template, or the name of the file without
+the extension, becomes the name of the stage, so it can be used as an argument
+to `--target`.
+
+When processing the template, `ci-operator` passes several parameters with
+information about the tests (e.g. the job name, which namespace it's operating
+into, URLs for images).  These parameters are described in its usage message.
+
+Logs from pods that fail will be fetched and stored if collection of artifacts
+is enabled (using the `--artifacts-dir` flag).  If part of the output is
+necessary even in case of success, it can be written to the artifact directory.
+
+Below is a sample job (extracted from `cluster/ci/config/prow/config.yaml`)
```

Comment:
chrome: you could use a relative link here to link directly to the file, e.g.
`extracted from [this config file](../cluster/ci/config/prow/config.yaml))`

Reply:
I think GItHub will generate the correct `href` with that server-side

Reply:
Changed just in case.